qiskit/dagcircuit/dagcircuit.py

```diff
@@ -1393,6 +1393,28 @@ def substitute_node(self, node, op, inplace=False):
             self._decrement_op(node.op)
         return new_node
 
+    def weakly_connected_components(self) -> List["DAGCircuit"]:
+        """Decompose the dag circuit into its weakly connected components."""
+        connected_components = rx.weakly_connected_components(self._multi_graph)
+
+        # Collect each disconnected subgraph
+        disconnected_subgraphs = []
+        for components in connected_components:
+            disconnected_subgraphs.append(self._multi_graph.subgraph(list(components)))
+
+        # Create new DAGCircuit objects from each of the rustworkx subgraph objects
+        decomposed_dags = []
+        for subgraph in disconnected_subgraphs:
+            new_dag = DAGCircuit()
+            new_dag.add_qubits(self.qubits)
+            for node in subgraph.nodes():
+                if not isinstance(node, DAGOpNode):
+                    continue
+                new_dag.apply_operation_back(node.op, node.qargs)
+            decomposed_dags.append(new_dag)
```

Comment:
I think handling global phase correctly here might be rather tricky.  Mathematically, I'd expect that `compose(dag.weakly_connected_compenents())` should restore the previous global phase, which means the individual phase angles should sum to the original circuit's.  But we don't _really_ have the information available to separate it out, I don't think, which means we run the risk of producing unexpected behaviour if somebody (for example) tries to apply some operation (e.g. control) on a subcomponent and tensor the graphs back together.

Those tricky questions aside, I think right now this completely erases global-phase information.  If we can't come up with better behaviour for the above, maybe that erasure is the best behaviour, but we should definitely make a big note of it in the docs if so.

Reply:
I've stuck with leaving the global phase erased and documenting it in the docstring. I'll add to release note as well